distributed/stealing.py

```diff
@@ -351,16 +347,22 @@ def maybe_move_task(level, ts, sat, idl, duration, cost_multiplier):
                             stealable.discard(ts)
                             continue
                         i += 1
-                        if not idle:
```

Comment:
It might still be nice to have the `if not idle: break` lines, just to avoid these checks when the cluster is under full load.